### PR TITLE
Make the "Become a Mentor" button responsive.

### LINKED ISSUE #71: The "Become a mentor" button is not properly responsive on mobile devices.
**Describe the bug**
The "Become a mentor" button is not properly responsive on mobile devices.

**To Reproduce**
1. Go to the homepage on a mobile device.
2. Scroll down to the "Become a mentor" button.
3. Notice that the button does not resize properly.

**Expected behavior**
The "Become a mentor" button should resize properly on mobile devices to ensure a good user experience.

**Screenshots**
mobileresponsive

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone 12]
 - OS: [e.g. iOS 14.4]
 - Browser Brave 
 - Version Version 1.61.114 Chromium: 120.0.6099.199 (Official Build) unknown (64-bit)

**Additional context**
This issue affects all pages where the "Become a mentor" button appears.

**Are you working on it?**

### PR DESCRIPTION
**Pull request by mayura-andrew**
## Purpose

The purpose of this PR is to fix #71 

## Purpose
The purpose of this PR is to make the "Become a Mentor" button responsive.

## Goals
This PR makes the "Become a Mentor" button responsive using Tailwind CSS. It also attempts to fix an issue where the application does not navigate to the `/mentor-registration` route when the button is clicked in mobile view.

## Approach
I updated the `handleMentorRegistration` function in the `MenuDrawer.tsx` file to navigate to the `/mentor-registration` route when the "Become a Mentor" button is clicked. However, this does not seem to be working as expected on mobile devices. I would appreciate if someone could look into this.

## Screenshots
[Screencast_20240113_161026.webm](https://github.com/sef-global/scholarx-frontend/assets/48531182/941639fc-8fe2-4fb0-bd63-0876f50b3fd8)

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Related PRs
N/A

## Test environment
OS - Fedora Linux 39 (KDE Plasma) x86_64
NodeJS - 20.10.0

## Learning
I learned about the importance of making UI elements responsive to ensure a good user experience on all devices. I also learned about the `navigate` function and how it can be used to programmatically navigate to different routes in a React application.

### CODE REVIEW

File: src/components/Layout/MenuDrawer/MenuDrawer.tsx [hunk 1]
```diff
@@ -26,6 +37,19 @@ const MenuDrawer: React.FC<MenuDrawerProps> = ({ openMenu, setOpenMenu }) => {
     setOpenMenu(false);
   };
 
+  const handleLoginModalOpen = (): void => {
+    setIsLoginModalVisible(true);
+  };
+
+  const handleMentorRegistration = (): void => {
+    console.log('Clicked');
```
Comment (anjula-sack) [inline src/components/Layout/MenuDrawer/MenuDrawer.tsx]:
```suggestion
```

File: src/components/Layout/MenuDrawer/MenuDrawer.tsx [hunk 2]
```diff
@@ -18,6 +24,11 @@ interface MenuDrawerProps {
 }
 
 const MenuDrawer: React.FC<MenuDrawerProps> = ({ openMenu, setOpenMenu }) => {
+  const [isLoginModalVisible, setIsLoginModalVisible] = useState(false);
```
Comment (anjula-sack) [inline src/components/Layout/MenuDrawer/MenuDrawer.tsx]: We don't need it use openMenu, setOpenMenu

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/Layout/MenuDrawer/MenuDrawer.tsx (hunk 2 above)
Comment (mayura-andrew) [reply src/components/Layout/MenuDrawer/MenuDrawer.tsx]: [Screencast_20240114_003851.webm](https://github.com/sef-global/scholarx-frontend/assets/48531182/4d317c0d-2b4f-4876-806a-3ac133e20679)

Fixed that issue :)

**Review by anjula-sack — APPROVED** [review]
Good job! @mayura-andrew

**Pull request merged**